### ISSUE TITLE
azure-native python package results in slow pulumi up times

### ISSUE THREAD

**Issue by MitchellGerdisch** [pulumi/pulumi-azure-native#738]
`pulumi up` for an azure-native python-based program takes an unexpectedly long time to run.

## Expected behavior

`pulumi up` for an azure-native python-based program should be a bit more efficient/quick when run.

## Current behavior

See steps below, but basically, step 7 shows a solid 30 seconds of a time gap between doing the `pulumi up` and the preview starting.

## Steps to reproduce

1. git clone https://github.com/pulumi/examples.git
2. cd examples/azure-py-webserver
3. pulumi up (it’ll install the dependencies and then fail due to confguration missing)
4. pulumi config set azure-native:location eastus
5. pulumi config set username webmaster
6. pulumi config set password —secret What3v3r1234
7. date; pulumi up; date

You will see a considerable delay (30-40s) between the `pulumi up` and the preview starting.

**Comment by t0yv0** [comment]
Here is a tuna trace of imports:

https://drive.google.com/file/d/1UYZ8Y16hKeYbrBWZuSx3l7U-Q0XHade2/view?usp=sharing

With `speed_test.py` as basically the imports only part of the pulumi program listed above:

```python
import base64
from pulumi import Config, Output, export
import pulumi_azure_native.compute as compute
import pulumi_azure_native.network as network
import pulumi_azure_native.resources as resources
```

To reproduce collect the import trace as:

```
env PYTHONPROFILEIMPORTTIME=1 time python speed_test.py 2> import_trace.log
pip install tuna
tuna import_trace.log
```

It does seem that 20s is a long time spent on import. Curiously many of the modules seem to be imported under various dated versions. I hypothesize only a small fraction of this code ends up being used in the example.

If this hypothesis holds, we can easily fix the issue by switching the SDK to lazy imports.

This [technique](https://stackoverflow.com/questions/42703908/how-do-i-use-importlib-lazyloader) may be applicable and compatible with Python 3.6. 

I will experiment a bit further and report here.

**Comment by t0yv0** [comment]
From the linked PR above:

> Warm pulumi preview on examples/azure-py-webserver goes from 27.51s to 3.55s. Cold preview spends time downloading packages and building venv, which is still slow, but we do not factor that time in here.

**Comment by Slaiyer** [comment]
Even with `pulumi-azure-native==1.3.0`, `pulumi preview` still takes ~1 min to run, with ~50s spent before the resource "graph" shows up - which I assume is related to this.

Also, while mypy memory consumption has gone down greatly, the time taken is still the same:
```
# mypy --show-error-codes .    
Success: no issues found in 7 source files                                                                                                                                                                   
~14m-15.4s 
```
Top-level pulumi imports:

1. `pulumi==3.1.0`
2. `pulumi-tls==4.0.0`
3. `pulumi-azure-native==1.3.0`
4. `pulumi-kubernetes==3.1.0`

I'll be glad to provide any more information that may be needed to help with this. Thanks!

**Comment by mikhailshilkov** [comment]
The fix is not included in 1.3.0, please try with 1.4.0 when it's out

**Comment by t0yv0** [comment]
Also mypy speedup proposal is still in review as it is breaking: https://github.com/pulumi/pulumi/issues/6837 - I see you upvoted it already, this is good.
